web/frontend/src/lib/components/robot-client.svelte

```diff
@@ -416,11 +423,11 @@ const start = () => {
   cancelTick = setAsyncInterval(tick, 500);
 };
 
-const connect = async (creds?: Credentials) => {
+const connect = async (creds?: Credentials, authEntity?: string) => {
   $connectionStatus = 'connecting';
 
   await $robotClient.connect({ 
-    authEntity: bakedAuth.authEntity, 
+    authEntity: authEntity ?? bakedAuth.authEntity,
     creds: creds ?? bakedAuth.creds,
     priority: 1 
   });
```

Comment:
AFAICT, this is where any delay would come from. Not sure where in the stack that is though and it might be baked into the `robotClient`. Doesn't look like any of this component is causing the delay.